Lombiq.HelpfulLibraries.AspNetCore/Security/ApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,72 @@
+﻿using Lombiq.HelpfulLibraries.AspNetCore.Security;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+
+using static Lombiq.HelpfulLibraries.AspNetCore.Security.ContentSecurityPolicyDirectives;
+using static Lombiq.HelpfulLibraries.AspNetCore.Security.ContentSecurityPolicyDirectives.CommonValues;
+
+namespace Microsoft.AspNetCore.Builder;
+
+public static class ApplicationBuilderExtensions
+{
+    /// <summary>
+    /// Adds a middleware that supplies <c>Content-Security-Policy</c> header. It may be further expanded by registering
+    /// services that implement <see cref="IContentSecurityPolicyProvider"/>.
+    /// </summary>
+    /// <param name="allowInline">If <see langword="true"/> then inline scripts and styles are permitted.</param>
+    public static IApplicationBuilder UseContentSecurityPolicyHeader(this IApplicationBuilder app, bool allowInline) =>
```

Comment:
Let's also have a parameter to supply further header values right away (and update the comment above too).

Reply:
My first thought was [this middleware's config](https://github.com/Lombiq/Lombiq-Tenants-Core/blob/90a26f19f8d8c2a833d17245358d01f455aa1809/src/Modules/Lombiq.Tenants.Core.Common/Services/ContentSecurityPolicyMiddleware.cs#L8), but that's actually indeed only interesting for one page.

But good point, let's leave it like that, but please comment why not having an easy way to add a global default is intentional, so we don't add it later.